sdk/js/src/Client.ts

```diff
@@ -442,7 +441,7 @@ export class DevCycleClient<
             }
             const config = await this.requestConsolidator.queue(updatedUser)
             if (user.isAnonymous || !user.user_id) {
```

Comment:
hmm does this logic actually make sense? why is it checking user and then saving updatedUser?

Reply:
I think that's intentional. It's checking whether the user had an id before being updated (which would add an anon user_id if there isnt one already). If it already had a user_id before being updated, then theres no need to save the anon user_id to storage because theres already one in there

Reply:
I think it's actually checking if the new user (passed in as `user`) is anonymous, and if so it's then saving the anon user id for this new user. But it's using `updatedUser` because the DVCPopulatedUser class is what generates the anon user id if it's not specified.